src/unitycatalog/ai/utils/function_processing_utils.py

```diff
@@ -110,6 +110,26 @@ def get_tool_name(func_name: str) -> str:
     return tool_name
 
 
+def construct_original_function_name(tool_name: str) -> str:
+    """
+    Args:
+        tool_name: The tool name in the form of `catalog__schema__function`.
```

Comment:
Can `catalog` contain `__`?

Reply:
I think it can (https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html#catalog-name), but I don't think we have a better way...
We do warn if the total name length is larger than 64. But we shouldn't prevent users from using the functions is the catalog/schema/function name contains `__`, because this function is only used in a helper function, and users can always customize based their own cases.

Reply:
what would happen if my function name is `foo.bar.my__func` and I use it as a tool?

Reply:
It's totally fine, but if you use `generate_tool_call_messages` helper function it would complain, as the converted tool name in OpenAI would be `foo__bar__my__func` (openAI tool doesn't allow `.` in the tool name), and we try to convert it back by splitting over `__` then it fails. While if you know that the original function name is `foo.bar.my__func` you don't need to rely on `generate_tool_call_messages`, and you can directly execute the function on your own